### PR TITLE
add response error handler and error page

### PR DESCRIPTION
**Pull request by maksimk99**
for more information see #9

### CODE REVIEW

**Comment by maksimk99** [comment]
Screenshot from 2019-09-26 12-51-27

**Comment by maksimk99** [comment]
return this page with an error message when the server response type is not 2xx

File: src/main/java/com/carhouse/handler/RestTemplateResponseErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * Handle client error and return error page.
+     *
+     * @param ex the exception
+     * @return the model and view
+     * @throws JsonProcessingException if can't convert json to object
+     */
+    @ExceptionHandler(HttpClientErrorException.class)
+    public ModelAndView handleClientError(final HttpClientErrorException ex) throws JsonProcessingException {
+        ExceptionJSONResponse response = new ObjectMapper().readValue(ex.getResponseBodyAsString(),
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/handler/RestTemplateResponseErrorHandler.java:27]: each time `new ObjectMapper()`. What for?

File: src/main/java/com/carhouse/handler/RestTemplateResponseErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                ExceptionJSONResponse.class);
+        ModelAndView model = new ModelAndView();
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/handler/RestTemplateResponseErrorHandler.java:29]: copypaste with `handleServerNotAvailableError`

**Review by aleksieiev — APPROVED** [review]

**Pull request merged**